### DISCUSSION TITLE
[Question]: How to include total quantity of an ingredient, but sub-quantities in steps

### DISCUSSION THREAD

**Discussion by shayaknyc** [cooklang/spec#79] [Ideas]
Hi! So, if I'm making a bunch of appetizers, for example, each serving of the appetizer will only have a portion of the total quantity of a particular ingredient, but I want to make sure the total amount needed for the recipe is captured in the ingredients. For example, if I need a total of 225g of ground beef, but each appetizer only needs 1/4 teaspoon of ground beef, when indicating the ingredient in the steps, I type:
```
Place about 0.25 teaspoons of @ground beef{225%grams} into each....
```
Which then renders as:
"Place about 0.25 teaspoons of 225 grams of ground beef...." - which just reads weirdly. Any ideas on how best to approach this?

**Comment by dubadub** [comment]
@shayaknyc this a perfectly valid use-case. We need to think about good syntax for that.

(Originally posted in issues in CookCLI, I moved that issue to the discussions as it's more related to language itself, not CLI).

**Reply by bubonicfred** [reply]
Perhaps some mark to not print the total amount in line? i.e. `0.25 teaspoons of @ground beef{^225%grams}` . Also I need to double-check how this renders in the webserver, I swear it doesn't print inline.

**Comment by jphastings** [comment]
This seems linked to [this discussion](https://github.com/cooklang/spec/discussions/96) — perhaps just referencing a quantity (without an ingredient) is the way to do this?

```
Place about %{0.25%teaspoons} of @ground beef{225%grams} into each
```

**Reply by FranGarc** [reply]
When you write a recipe  you may mention one or more ingredients multiple times, but since it's the same amount, you just write @ingredient{amount} the first time, and just @ingredient the followng ones. When you use different amounts of a same ingredients, you go @ingredient{amount1} and @ingredient{amount2}.

Think of a cupcake recipe that includes the buttercream or whatever filling you like. Both things have _at least_ sugar in common, so it would go like

> Make the batter with @sugar{400g}. Mix the sugar with the flour in a bowl separate from the wet ingredients.
> --- something else---
> Make the filling with @butter{200g} @sugar{300g}. Whip the butter with the sugar and the @cinnamon until creamy

Ideally, the CLI would  list the ingredient as

> Sugar: **700g**

**Comment by FreeMasen** [comment]
I believe this would also be an appropriate place to bring up how to handle division of ingredients that aren't entirely measurable. For example, splitting an egg into the egg white and the egg yolk?

**Reply by shayaknyc** [reply]
Oh, that would be cool too! Though, when I need to do that, I usually just add egg yolk and egg white as separate ingredients.....

**Reply by FreeMasen** [reply]
I wonder if there might be space for a sort of transformation, that would end up generating something like an intermediate ingredient, not dissimilar to the [intermediate-preparations](https://github.com/cooklang/cooklang-rs/blob/main/extensions.md#intermediate-preparations) extension. This might end up taking an existing ingredient and generating multiple intermediate ingredients. Not exactly sure what the syntax should be but my initial thought was  something like

```
split @eggs{2}->(@egg yolk{2}, @egg white{2})
```

The arrow could maybe be swapped for another keyword like `into` or some other sigil?

**Comment by hilli** [comment]
Isn't this solved by having servings in the frontmatter and specifying what the teaspoon of beef amounts to in `g`?

```yaml
---
servings: 30
---
```
```
Place about 0.25 teaspoons of @ground beef{8%g} into each....
```

Then the rendering will produce the 240g (8g*30) in the ingredients list.
